### ISSUE TITLE
ReactReduxSpa template logs strange messages when there's a client-side 404

### ISSUE THREAD

**Issue by SteveSandersonMS** [aspnet/JavaScriptServices#74]
The ReactReduxSpa template's server-side prerendering code works well when the client-side routes are matched, but if they fail to match, it logs errors like this:

```
[Node] Warning: Failed propType: Required prop `router` was not specified in `RouterContext`.
[Node] Warning: Failed propType: Required prop `location` was not specified in `RouterContext`.
[Node] Warning: Failed propType: Required prop `routes` was not specified in `RouterContext`.
[Node] Warning: Failed propType: Required prop `params` was not specified in `RouterContext`.
[Node] Warning: Failed propType: Required prop `components` was not specified in `RouterContext`.
[Node] Warning: [react-router] `<RouterContext>` expects a `router` rather than a `history`
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[0]
      An unhandled exception has occurred while executing the request
System.Exception: Call to Node module failed with error: TypeError: Cannot read property 'listenBeforeLeavingRoute' of undefined
    at getChildContext (NodeServices/templates/ReactReduxSpa/node_modules/react-router/lib/RouterContext.js:80:35)
    at ReactCompositeComponentMixin._processChildContext (/NodeServices/templates/ReactReduxSpa/node_modules/react/lib/ReactCompositeComponent.js:413:53)
```

This is confusing because it doesn't say anywhere what the underlying cause was (a request that doesn't match any client-side route) and looks worse than it is.

Somehow, `boot-server.tsx` needs to handle this situation better.

**Comment by laskoviymishka** [comment]
This expected behaivior, see _https://github.com/reactjs/react-router/issues/862_. To prevent such strange logs may be it better to adopt this code to react redux template? 
In a fact we need to check does route really exist (i.e. not 404) and only after that try to render html string.

**Comment by SteveSandersonMS** [comment]
> To prevent such strange logs may be it better to adopt this code to react redux template? 

Can you clarify what you mean here? I don't understand what code you're referring to.

> In a fact we need to check does route really exist (i.e. not 404)

Is there a way to do that?

**Comment by laskoviymishka** [comment]
@SteveSandersonMS i refer to initial issue code sample.
It need to add just:
```
if(routes.some(route => route.path == params.location){
   // regular operation
}else{
   // some 404 action
}
```

**Comment by SteveSandersonMS** [comment]
Fixed with 2ffc31a7c6e174506392461b2f205933c8368911
